aptos-move/framework/src/built_package.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error_map::generate_error_map;
+use crate::natives::code::{ModuleMetadata, PackageMetadata, UpgradePolicy};
+use aptos_types::account_address::AccountAddress;
+use clap::Parser;
+use move_deps::move_core_types::errmap::ErrorMapping;
+use move_deps::move_package::compilation::compiled_package::CompiledPackage;
+use move_deps::move_package::BuildConfig;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+
+/// Represents a set of options for building artifacts from Move.
+#[derive(Debug, Clone, Parser, Serialize, Deserialize)]
+pub struct BuildOptions {
+    #[clap(long, default_value = "true")]
+    pub with_srcs: bool,
+    #[clap(long, default_value = "true")]
+    pub with_abis: bool,
+    #[clap(long, default_value = "true")]
+    pub with_source_maps: bool,
+    #[clap(long, default_value = "true")]
+    pub with_error_map: bool,
```

Comment:
this won't work, bools cannot be passed in via cli... just trying now with the aptos command line and the publish flow.

Reply:
> Considering the new devx, it would be great to add some sort of readme to the framework directory that indicates the new development flow

We should really do this, but it need to postponed for some days. The docs folks need is on the CLU level. which I plan to write.

Reply:
> Also would be great to have some sort of test that ensures that the committed mrb files are up to date.

It's there!

Reply:
> Finally, I think you'll need to update the build flows for docker:

Oh boy. Is there someone who owns this docker code? What is the objective of this code? Is it in regular use so we need keep it working to land this PR? 

@geekflyer

https://github.com/aptos-labs/aptos-core/blob/bcc1bd18937b7c4c69a4b47c64413a92adea7ded/docker/rust-all.Dockerfile#L149

Reply:
[Oh Github UI ... I could *not* comment on the individual comments in the stream. So all replies landed on the last item in the stream.]

Reply:
> > Finally, I think you'll need to update the build flows for docker:
> 
> Oh boy. Is there someone who owns this docker code? What is the objective of this code? Is it in regular use so we need keep it working to land this PR?
> 
> @geekflyer
> 
> https://github.com/aptos-labs/aptos-core/blob/bcc1bd18937b7c4c69a4b47c64413a92adea7ded/docker/rust-all.Dockerfile#L149

Those lines in the docker build files have been touched last ~5 months ago in:
https://github.com/aptos-labs/aptos-core/pull/72/files#diff-cde3368c11183b2f9fb91b7fa3063d604f3170c12afe1efb8b94dd607b979885R49-R55
and
https://github.com/aptos-labs/aptos-core/commit/aa4684d1f634d6b7698d508581068c4eb75ef0d6#diff-cde3368c11183b2f9fb91b7fa3063d604f3170c12afe1efb8b94dd607b979885R38

@davidiw  and @sherry-x  can probably best answer the objective of this code.
Personally I'd say say just remove them and apply the label `CICD:run-e2e-test` and hope things pass.

Reply:
I think this code is mainly for running release to generate the mv files and put them into a target folder that is eventually used to generate the genesis.